contracts/BitGuildMarketplace.sol

```diff
@@ -0,0 +1,334 @@
+
+    modifier onlyWhitelisted(address _contract) {
+        require(Whitelist.isWhitelisted(_contract), "Contract not in whitelist.");
+        _;
+    }
+
+    // @dev fall back function
+    function () external payable {
+        revert();
+    }
+
+    // @dev Retrieve hashkey to view listing
+    function getHashKey(address _contract, uint _tokenId) public pure returns(bytes32 key) {
+        key = _getHashKey(_contract, _tokenId);
+    }
+
+    // ===========================================
+    // Fee functions (from fee provider contract)
+    // ===========================================
+    // @dev get default fee
+    function getFee(uint _price) public view returns(uint percent, uint fee) {
+        (percent, fee) = FeeProvider.getFee(_price);
+    }
+
+    // @dev get custom fees
+    function getFee(uint _price, address _buyer, address _seller, address _token) public view returns(uint percent, uint fee) {
+        (percent, fee) = FeeProvider.getFee(_price, _buyer, _seller, _token);
+    }
+
+    // ===========================================
+    // Seller Functions
+    // ===========================================
+    // Deposit Item
+    // @dev deprecated callback (did not handle operator). added to support older contracts
+    function onERC721Received(address _from, uint _tokenId, bytes _extraData) external returns(bytes4) {
+        _deposit(_from, msg.sender, _tokenId, _extraData);
+        return ERC721_RECEIVED_OLD;
+    }
+
+    // @dev expected callback (include operator)
+    function onERC721Received(address _operator, address _from, uint _tokenId, bytes _extraData) external returns(bytes4) {
+        _deposit(_from, msg.sender, _tokenId, _extraData);
+        return ERC721_RECEIVED;
+    }
+
+    // @dev Extend item listing: new expiry = current expiry + defaultExpiry
+    // @param _contract whitelisted contract
+    // @param _tokenId  tokenId
+    function extendItem(address _contract, uint _tokenId) public onlyWhitelisted(_contract) returns(bool) {
+        bytes32 key = _getHashKey(_contract, _tokenId);
+        address seller = listings[key].seller;
+
+        require(seller == msg.sender, "Only seller can extend listing.");
+        require(listings[key].expiry > 0, "Item not listed.");
+
+        listings[key].expiry = now + defaultExpiry;
+
+        emit LogListingExtended(seller, _contract, _tokenId, listings[key].createdAt, listings[key].expiry);
+
+        return true;
+    }
+
+    // @dev Withdraw item from marketplace back to seller
+    // @param _contract whitelisted contract
+    // @param _tokenId  tokenId
+    function withdrawItem(address _contract, uint _tokenId) public onlyWhitelisted(_contract) {
+        bytes32 key = _getHashKey(_contract, _tokenId);
+        address seller = listings[key].seller;
+
+        require(seller == msg.sender, "Only seller can withdraw listing.");
+
+        if (!allowWithdrawBeforeExpiry) {
```

Comment:
I think safe to remove the param and the logic here, we don't wanna lock the item before it expires